---
title: "Single Cell RNA-Sequencing REMI Tutorial"
author: "Alice Yu"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Single Cell RNA-Sequencing REMI Tutorial}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r}
source("../R/load_functions.R")
source("../R/load_subfunctions.R")
load("../data/curr.lr.filt.rda")
load("../data/g.biogrid.rda")
load("../data/pathway.genelist.rda")
load("../data/pathway.genes.rda")
```

Download data from: https://figshare.com/s/871cb53e8fb5206bf8d0

Dataset is from Laughney et al. Regenerative lineages and immune-mediated pruning in lung cancer metastasis (2020). Data has been integrated and scaled.

Read in dataset formatted as a Seurat object
```{r}
seurat.obj <- readRDS("~/Downloads/laughneytumor_forREMI.rds")
```

## Constructing a REMI network on subset of Laughney et al's single-cell RNA-sequencing dataset
- Current run time ~ 20 minutes, working on speeding up the run time
```{r}
#Set sample.col as column listing sample IDs
#Set celltype.col as column listing cell type category of interest
# Filter = TRUE removes low expressed ligand and receptors from analysis
# Threshold = expression level cut-off
# Expthres = percent expressed per cell type cut-off
data.obj <- setupSingleCell(seurat.obj, 
                          sample.col = "Sample.ID",
                          celltype.col = "CELL_TYPE",
                          filter=T,
                          assay="integrated",
                          use.celltypes = c("EPITHELIAL", "FIBROBLAST"),
                          expthres = 0.2)

# Run REMI using default parameters
remi.obj <- remi(data.obj)
```

Viewing REMI network
- Recommend removing edges with lower weights (i.e. < 0.1)
```{r}
head(remi.obj$interactome)
```

Plotting REMI Circo Plot
```{r}
library(RColorBrewer)
# Setting colors for each cell type
grid.col = c(brewer.pal(8, "Set2"), brewer.pal(8, "Set1"))
names(grid.col) <- data.obj$cellmarkers
REMIPlot(remi.obj, grid.col=grid.col, legend=F)
```
